openff/interchange/components/toolkit.py

```diff
@@ -105,35 +108,98 @@ def _check_electrostatics_handlers(force_field: "ForceField") -> bool:
     return False
 
 
-def _simple_topology_from_openmm(openmm_topology: "openmm.app.Topology") -> Topology:
-    """Convert an OpenMM Topology into an OpenFF Topology consisting **only** of so-called `_SimpleMolecule`s."""
+def _simple_topology_from_openmm(
+    openmm_topology: "openmm.app.Topology",
+    system: openmm.System | None = None,
+) -> Topology:
+    """
+    Convert an OpenMM Topology into an OpenFF Topology consisting **only** of so-called `_SimpleMolecule`s.
+
+    Arguments
+    ---------
+    openmm_topology: openmm.app.Topology
+        The OpenMM Topology to convert.
+    system: openmm.System, optional
+        The OpenMM System associated with the topology. Only needed if there are virtual sites in the topology.
+    """
     # TODO: Splice in fully-defined OpenFF `Molecule`s?
 
     graph = networkx.Graph()
 
+    virtual_sites: list[ImportedVirtualSiteKey] = list()
+
+    # map indices of OpenMM system with virtual site to topology indices, or virtual site keys,
+    # in associated OpenFF topology, since stripping out virtual sites offsets particle indices
+    openmm_openff_particle_map: dict[int, int | ImportedVirtualSiteKey] = dict()
+
     # TODO: This is nearly identical to Topology._openmm_topology_to_networkx.
     #  Should this method be replaced with a direct call to that?
```

Comment:
Is this comment still up-to-date? :upside_down_face:

Reply:
Nope, nice catch